samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -150,6 +152,13 @@
    */
   private final Optional<StreamRegexMonitor> inputStreamRegexMonitor;
 
+  /**
+   * Container placement action dispatcher and util
+   */
+  private final ContainerPlacementUtil containerPlacementUtil;
+  private final ContainerPlacementHandler containerPlacementHandler;
+  private final Thread containerPlacementHandlerThread;
```

Comment:
// containerPlacement thread that reads and writes placement-requests and responses to the metastore